src/ray/gcs/tables.cc

```diff
@@ -273,6 +273,38 @@ Status ProfileTable::AddProfileEventBatch(const ProfileTableData &profile_events
                 });
 }
 
+Status ActorTable::AppendDataAt(const ActorID &actor_id,
```

Comment:
I think it would make sense to expose fewer arguments for this method and make the logic more specific to the actor table. Specifically, I would at minimum remove `node_manager_id`, which you can get from `ActorTable::client_`, and `log_length`, which you can compute from `max_reconstructions` and `remaining_reconstructions`. Ideally, we would just pass in a `const ActorRegistration &`, which already contains most of the arguments here.

I would probably also rename the method to something like `UpdateActorState`.

Also, I would prefer if we passed in a callback (of type `ActorTable::WriteCallback`) instead of `error_on_failure`, to make it clear to the caller what will happen if the operation fails. The `bool` argument is a bit opaque.